docs/source/examples/Widget Custom.ipynb

```diff
@@ -0,0 +1,785 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "source": [
+    "[Index](Index.ipynb) - [Back](Widget Styling.ipynb) - [Next](Widget Asynchronous.ipynb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# Building a Custom Widget -  Email widget"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The widget framework is built on top of the Comm framework (short for communication).  The Comm framework is a framework that allows the kernel to send/receive JSON messages to/from the front end (as seen below).\n",
+    "\n",
+    "![Widget layer](images/WidgetArch.png)\n",
+    "\n",
+    "To create a custom widget, you need to define the widget both in the browser and in the Python kernel.\n",
+    "\n",
+    "This tutorial shows how to build a simple email widget using the TypeScript widget cookiecutter: https://github.com/jupyter-widgets/widget-ts-cookiecutter"
```

Comment:
I'd put the screenshot of the completed widget here as well. So that people have a sense of what they'll get at the end of the tutorial

Reply:
Sounds good, thanks!